.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,233 @@
+      If enabled, Camunda Nexus will be configured and used as a mirror for Maven repositories
+      Camunda Nexus is automatically disabled for repositories that do not belong to the camunda organization, such as forks.
+    required: false
+    default: "true"
+  dockerhub:
+    description: |
+      If enabled, logs into DockerHub with a CI account.
+    required: false
+    default: "false"
+  dockerhub-readonly:
+    description: |
+      If enabled, logs into DockerHub with a read-only CI account.
+      Useful to avoid being rate-limited when using an anonymous account from GitHub-managed runners.
+    required: false
+    default: "false"
+  java-distribution:
+    description: Java distribution to use
+    required: false
+    default: temurin
+  java-version:
+    description: JDK version to install
+    required: false
+    default: "21"
+  maven-cache-key-modifier:
+    description: A modifier to use for maven cache key
+    required: false
+    default: "shared"
+  maven-mirrors:
+    description: |
+      JSON list of extra Maven mirrors.
+      It will be merged with Camunda Nexus settings (if enabled ), with precedence."
+    required: false
+    default: '[]'
+  maven-servers:
+    description: |
+      JSON list of extra Maven servers.
+      It will be merged with Camunda Nexus settings (if enabled), with precedence."
+    required: false
+    default: '[]'
+  maven-version:
+    description: |
+      Maven version to be used by the wrapper.
+      If not set, default version in .mvn/wrapper/maven-wrapper.properties will be used.
+    required: false
+  time-zone:
+    description: |
+      Time zone to set for the build environment. Use TZ identifiers, e.g., 'Europe/Berlin'.
+      https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+    required: false
+  vault-address:
+    description: Vault URL to retrieve secrets from
+    required: false
+  vault-role-id:
+    description: Vault Role ID to use
+    required: false
+  vault-secret-id:
+    description: Vault Secret ID to use
+    required: false
+
+outputs: {}
+
+runs:
+  using: composite
+  steps:
+  - name: Preliminary checks
+    id: checks
+    env:
+      DOCKERHUB: ${{ inputs.dockerhub }}
+      DOCKERHUB_READONLY: ${{ inputs.dockerhub-readonly }}
+      # Paths to Vault secrets for DockerHub accounts
+      DOCKERHUB_VAULT_PATHS: >-
+        {
+          "camunda": {
+            "path": "secret/data/products/camunda/ci/github-actions"
+          },
+          "operate": {
+            "path": "secret/data/products/operate/ci/github-actions"
+          },
+          "optimize": {
+            "path": "secret/data/products/optimize/ci/camunda-optimize"
+          },
+          "tasklist": {
+            "path": "secret/data/products/tasklist/ci/tasklist"
+          },
+          "zeebe": {
+            "path": "secret/data/products/zeebe/ci/zeebe"
+          }
+        }
+      REPO_OWNED_BY_CAMUNDA: ${{ github.repository_owner == 'camunda' }}
```

Comment:
❓ This expression is very lean and elegant way to check. Is it available in all contexts (on fork PRs etc)? The setup-zeebe action used a different expression for this kind of check (`github.event.name == 'pull_request' && github.event.pull_request.head.repo.full_name == 'camunda/camunda'`) that reads more (over-?)complicated.

Reply:
I was inspired by https://github.com/orgs/community/discussions/26409, i did not tested it on a real fork but I guess it should work for all type contexts.

The original check from `setup-zeebe` was to fail if an internal PR does not define GitHub secrets for Vault. I think it is not necessary (but still provide nice error log) because it will fail anyway on the Vault step ... .
```yaml
    - name: Secrets check
      if: |
        github.event.name == 'pull_request'
        && github.event.pull_request.head.repo.full_name == 'camunda/camunda'
        && (inputs.secret_vault_address == ''
        || inputs.secret_vault_roleId == ''
        || inputs.secret_vault_secretId == '')
      shell: bash
      run: |
        echo "We will fail the action if it's an internal PR without the secrets. If this occurs it indicates a configuration failure that needs to be fixed."
        exit 1
```

I will check different contexts to validate the approach.

Reply:
Sounds good to me 👍🏼

Reply:
It does not work as I expected 😢 

I created a test fork, see this workflow run and job https://github.com/camunda/infra-global-github-actions/actions/runs/11727212631/job/32667727156

<img width="857" alt="image" src="https://github.com/user-attachments/assets/d3b00f87-b95e-410a-9523-0130a609d37b">

I need to rely on  `github.event.pull_request.head.repo.full_name` to detect workflow runs for fork PRs.
Really good catch!!

Reply:
I switch from a repo-owned-by-camunda check to a is-fork check which is more clear btw.